pkg/manifests/manifests.go

```diff
@@ -189,17 +233,30 @@ func newIngressControllerClusterRoleBinding(conf *config.Config) *rbacv1.Cluster
 			Name:     IngressControllerName,
 		},
 		Subjects: []rbacv1.Subject{{
-			Kind:      "ServiceAccount",
+			Kind:      serviceAccountKind,
 			Name:      IngressControllerName,
 			Namespace: conf.NS,
 		}},
 	}
 }
 
+func getIngressControllerService(conf *config.Config) *corev1.Service {
+	serviceObj := newIngressControllerService(conf)
+	if conf.DNSZoneDomain != "" && conf.DNSZonePrivate {
+		annotations := map[string]string{
+			"service.beta.kubernetes.io/azure-load-balancer-internal": "true",
+			"external-dns.alpha.kubernetes.io/hostname":               fmt.Sprintf("loadbalancer.%s", conf.DNSZoneDomain),
+			"external-dns.alpha.kubernetes.io/internal-hostname":      fmt.Sprintf("clusterip.%s", conf.DNSZoneDomain),
+		}
+		serviceObj.ObjectMeta.Annotations = annotations
```

Comment:
Update specific keys instead of replacing the entire map pls

Reply:
prior to my transformation the serviceObj doesn't have a map at serviceObj.ObjectMeta.Annotations - there's only a name, namespace, and labels

Reply:
following discussion, I will add a blank map in the original annotations position.